### PR TITLE
Added sealedtypes

### PR DESCRIPTION
**Pull request by cayhorstmann**

### CODE REVIEW

File: site/content/features/sealedtypes.md [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+---
+
+
+Java 15 provides a preview of “sealed types”—types with a fixed set of direct subtypes. This feature allows for accurate modeling of type hierarchies that should not be open to arbitrary inheritance, and it allows the compiler to check for exhaustive pattern matching.
+
+## Controlling Subtypes
+
+Unless a class is declared `final`, anyone can form a subclass of it. What if you want to have more control? For example, suppose you feel the need to write your own JSON library because none of the ones out there do exactly what you need.
+
+The JSON standard says that a JSON value is an object, array, string, number, Boolean, or `null`. The Java language had, until now, no way of expressing that a `JSONValue` should be exactly one of those six types.
```
Comment (marchof) [inline site/content/features/sealedtypes.md:16]: Is "Boolean" upper case intentionally?

File: site/content/features/sealedtypes.md [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+
+```
+public final class JSONNull extends JSONValue {
+   // . . .
+
+}
+```
+
+```
+public class Sandbox {
```
Comment (marchof) [inline site/content/features/sealedtypes.md:82]: I wonder whether these source snippets should have gone into a sandbox tag?

**Comment by marchof** [comment]
Do you have a chance to limit the json.png image to 800px? This is the width of the content column on javaalmanac.io, currently the image exceeds the right border.

**Comment by marchof** [comment]
The JEP and the title of this article is "Sealed classes". For consistency shouldn't the files also be called `sealedclasses`?

File: site/content/features/sealedtypes.md [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+   public static void main(String[] args) {
+      System.out.println("Move along, nothing to see here...");
+   }
+}
+```
+
+In contrast, `non-sealed` is a keyword. Obviously, you cannot use it as an identifier since it contains a `-` character. In fact, `non-sealed` is the second keyword that contains a character that isn't a lowercase letter. The first one is `_`, since Java 9. I am sure
```
Comment (marchof) [inline site/content/features/sealedtypes.md:187]: What means "I am sure" here?

File: site/content/features/sealedtypes.md [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+   private Path path;
+   public Path getPath() { return path; }
+
+   // Direct subclasses in same source file
+   public static final class Directory extends DirectoryEntry {
+      private Directory() {}
+      public Stream<DirectoryEntry> entries() {
+         try {
+            return Files.list(getPath()).map(DirectoryEntry::of);
+         } catch (AccessDeniedException ex) {
```
Comment (marchof) [inline site/content/features/sealedtypes.md:229]: What about simply declaring IOException to make the example more concise?

Comment (marchof) [reply site/content/features/sealedtypes.md:229]: Ah, doesn't work as you can't handle the checked exception in the `Consumer` API.

File: site/content/features/sealedtypes.md [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+   public static void main(String[] args) {
+      var positiveIntegers = new IteratedSeq(x -> x + 1, 1);
+      System.out.println(sum(positiveIntegers, 100));
+   }
+}
+```
+
+# Records and Enums
+
+A sealed class can be extended by a record, which is implicitly `final`. Consider the classic example of a Lisp-style list:
```
Comment (marchof) [inline site/content/features/sealedtypes.md:370]: AFAIK records cannot extend classes. Should this read "A sealed interface can be extended by a record"

**Comment by marchof** [comment]
What about a "References" section at the end with the link to the JEP (like in the other articles)?

File: site/content/features/sealedtypes.md (hunk 1 above)
Comment (cayhorstmann) [reply site/content/features/sealedtypes.md:16]: Yes, it's an adjective derived from a person's name. https://english.stackexchange.com/questions/4481/should-the-word-boolean-be-capitalized. I'll change it to Boolean value.

File: site/content/features/sealedtypes.md (hunk 2 above)
Comment (cayhorstmann) [reply site/content/features/sealedtypes.md:82]: Yes, my bad. Something went wrong in the conversion from XHTML to Markdown. I just fixed that.

File: site/content/features/sealedtypes.md (hunk 3 above)
Comment (cayhorstmann) [reply site/content/features/sealedtypes.md:187]: Removed it

File: site/content/features/sealedtypes.md (hunk 5 above)
Comment (cayhorstmann) [reply site/content/features/sealedtypes.md:370]: Fixed to "A sealed interface can be implemented by a record"

File: site/content/features/sealedtypes.md (hunk 1 above)
Comment (marchof) [reply site/content/features/sealedtypes.md:16]: Ups, I wasn't aware of this. So please leave it like it is. (Another learning for me)

**Comment by cayhorstmann** [comment]
> Do you have a chance to limit the json.png image to 800px? This is the width of the content column on javaalmanac.io, currently the image exceeds the right border.

Done

**Comment by cayhorstmann** [comment]
> The JEP and the title of this article is "Sealed classes". For consistency shouldn't the files also be called `sealedclasses`?

Last time, I called all the files Sandbox.java, so I am consistent :-) Let me know if you want them renamed. I don't care either way.

**Comment by marchof** [comment]
> > The JEP and the title of this article is "Sealed classes". For consistency shouldn't the files also be called `sealedclasses`?
> 
> Last time, I called all the files Sandbox.java, so I am consistent :-) Let me know if you want them renamed. I don't care either way.

I'm just talking about the filename of the article itself.

**Comment by cayhorstmann** [comment]
> > > The JEP and the title of this article is "Sealed classes". For consistency shouldn't the files also be called `sealedclasses`?
> > 
> > 
> > Last time, I called all the files Sandbox.java, so I am consistent :-) Let me know if you want them renamed. I don't care either way.
> 
> I'm just talking about the filename of the article itself.

The JEP is obviously misnamed. I'd prefer to change the title.

**Comment by marchof** [comment]
> > > > The JEP and the title of this article is "Sealed classes". For consistency shouldn't the files also be called `sealedclasses`?
> > > 
> > > 
> > > Last time, I called all the files Sandbox.java, so I am consistent :-) Let me know if you want them renamed. I don't care either way.
> > 
> > 
> > I'm just talking about the filename of the article itself.
> 
> The JEP is obviously misnamed. I'd prefer to change the title.

Good idea!

**Pull request merged**